bananas_api/new_upload/extract.py

```diff
@@ -0,0 +1,114 @@
+import tarfile
+import os
+import secrets
+import zipfile
+
+TAR_STORAGE_PATH = "data/tar"
+
+
+def _find_root_folder(info_list, get_name=None, is_file=None):
+    """
+    Tar-files are often made of a whole folder. This means that we would
+    prefix all files with that folder name, which in 99% of the cases
+    won't be the expected outcome. So first do a scan, to see if there
+    are any files or more than one directory in the root folder. If not,
+    skip the root-folder while extracting.
+
+    To keep things more the same, do this also for any other format.
+    """
+
+    root_folder = None
+
+    for info in info_list:
+        if not is_file(info):
+            continue
+
+        if root_folder is None:
+            root_folder = get_name(info).split("/")[0]
```

Comment:
If the archive contains a single file, it will consider the filename as root folder.

Reply:
The line after would catch that. Files don't hav a / in them, so it will be set to None again. I think. :D